File: src/main/java/com/csl/parking/Car.java
package com.csl.parking;

import java.time.Duration;
import java.time.LocalDateTime;

public class Car {
    private String id;
    private LocalDateTime enterTime;
    private LocalDateTime leaveTime;

    public Car(String id, LocalDateTime enterTime) {
        this.id = id;
        this.enterTime = enterTime;
    }

    public void leave() {
        leaveTime = LocalDateTime.now();
    }

    public Duration getDuration() {
        Duration duration = Duration.between(enterTime, leaveTime);
        return  duration;
    }

    public void setLeaveTime(LocalDateTime leaveTime) {
        if (leaveTime.isAfter(enterTime))
            this.leaveTime = leaveTime;
    }
}

File: src/main/kotlin/com/csl/parking/Car.kt
package com.csl.parking

import java.time.Duration
import java.time.LocalDateTime

class CarKt(id: String, enterTime: LocalDateTime) {

    var id:String=""
    var enterTime:LocalDateTime
    var leaveTime:LocalDateTime = LocalDateTime.now()
        set(value) {
            if (leaveTime.isAfter(enterTime))
                field = value
        }

    init {
        this.id = id
        this.enterTime = enterTime
    }

    fun getDuration():Long {
        return Duration.between(enterTime, leaveTime).toMinutes()
    }



}